src/ts/src/common/components/library/data_table/index.tsx

```diff
@@ -1,66 +1,73 @@
 import React, { FunctionComponent } from "react";
 import { Table } from "semantic-ui-react";
-import { ReactComponent as TreeIcon } from "src/common/icons/PhyloTree.svg";
-import { ReactComponent as SampleIcon } from "src/common/icons/Sample.svg";
+
 import style from "./index.module.scss";
 
 interface Props {
-  data?: BioinformaticsData[];
-  headers: Header[];
+    data?: Record<string | number, JSONPrimitive>[];
+    headers: Header[];
+    renderer?: CustomRenderer;
 }
 
-const ICONS: Record<string, JSX.Element> = {
-  Sample: <SampleIcon className={style.icon} />,
-  Tree: <TreeIcon className={style.icon} />,
-};
-
 const UNDEFINED_TEXT = "---";
 
-const DataTable: FunctionComponent<Props> = ({ data = [], headers }: Props) => {
-  const indexingKey = headers[0].key;
-
-  // render functions
-  const headerRow = headers.map((column: Header) => (
-    <Table.HeaderCell key={column.key}>
-      <div className={style.headerCell}>{column.text}</div>
-    </Table.HeaderCell>
-  ));
-
-  const sampleRow = (item: BioinformaticsData): Array<JSX.Element> => {
-    return headers.map((column, index) => {
-      let displayData = item[column.key];
-      let icon: JSX.Element | null = null;
-      if (displayData === undefined) {
+function defaultCellRenderer({ value }: CustomTableRenderProps): JSX.Element {
+    let displayData;
+    if (value === undefined) {
         displayData = UNDEFINED_TEXT;
-      }
-      if (index === 0) {
-        icon = ICONS[item.type];
-      }
-      return (
-        <Table.Cell key={`${item[indexingKey]}-${column.key}`}>
-          <div className={style.cell}>
-            {icon}
-            {displayData}
-          </div>
-        </Table.Cell>
-      );
-    });
-  };
+    } else {
+        displayData = value;
+    }
+    return <div className={style.cell}>{displayData}</div>;
+}
 
-  function tableRows(data: BioinformaticsData[]): Array<JSX.Element> {
-    return data.map((item) => (
-      <Table.Row key={`${item[indexingKey]}`}>{sampleRow(item)}</Table.Row>
+const DataTable: FunctionComponent<Props> = ({
+    data = [],
+    headers,
+    renderer,
```

Comment:
we can add default value to renderer here, so we don't need to reassign it later!

E.g.,

```tsx
const DataTable: FunctionComponent<Props> = ({
    data = [],
    headers,
    renderer = defaultCellRenderer, // <-- this!
}: Props) => {
```